x/auction/simulation/operations.go

```diff
@@ -60,41 +83,56 @@ func SimulateMsgPlaceBid(authKeeper auth.AccountKeeper, keeper auction.Keeper) s
 			return true // found valid pair
 		})
 		if !found {
-			return simulation.NewOperationMsgBasic(auction.ModuleName, "no-operation (no valid auction and bidder)", "", false, nil), nil, nil
+			return simulation.NewOperationMsgBasic(types.ModuleName, "no-operation (no valid auction and bidder)", "", false, nil), nil, nil
+		}
+
+		bidderAcc := ak.GetAccount(ctx, bidder.Address)
+		if bidderAcc == nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, nil
 		}
 
 		// pick a bid amount for the chosen auction and bidder
-		amount, _ := generateBidAmount(r, openAuction, bidder, blockTime)
+		amount, err := generateBidAmount(r, openAuction, bidderAcc, blockTime)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
 
 		// create a msg
-		msg := auction.NewMsgPlaceBid(openAuction.GetID(), bidder.GetAddress(), amount)
-		if err := msg.ValidateBasic(); err != nil { // don't submit errors that fail ValidateBasic, use unit tests for testing ValidateBasic
-			return noOpMsg, nil, fmt.Errorf("expected msg to pass ValidateBasic: %s", msg.GetSignBytes())
+		msg := types.NewMsgPlaceBid(openAuction.GetID(), bidder.Address, amount)
+
+		spendable := bidderAcc.SpendableCoins(ctx.BlockTime())
+		fees, err := simulation.RandomFees(r, ctx, spendable)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
+
+		tx := helpers.GenTx(
+			[]sdk.Msg{msg},
+			fees,
+			helpers.DefaultGenTxGas,
+			chainID,
+			[]uint64{bidderAcc.GetAccountNumber()},
+			[]uint64{bidderAcc.GetSequence()},
+			bidder.PrivKey,
+		)
+
+		_, result, err := app.Deliver(tx)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
```

Comment:
Will returning an error here stop the simulation?

Reply:
correct 👍